### PR TITLE
Suppress readability-redundant-member-init

### PR DESCRIPTION
**Pull request by danakj**
This initializes the DriverResult::per_file_success field explicitly with `= {}` in order to encode that DriverResult can be constucted via aggregate initialization while omitting the per_file_success field. This prevents -Wmissing-designated-field-initializers from firing in newer clang versions when constructing DriverResult like:
```
return {.success = false};
```

Newer clang-tidy warns that the `= {}` is redundant however it is not, as its marking which fields need to be explicitly initialized. So we suppress it.

### CODE REVIEW

File: .clang-tidy [hunk 1]
```diff
@@ -46,6 +46,7 @@ Checks:
   -readability-function-cognitive-complexity, -readability-else-after-return,
   -readability-identifier-length, -readability-implicit-bool-conversion,
   -readability-magic-numbers, -readability-make-member-function-const,
+  -readability-redundant-member-init,
```
Comment (jonmeow) [inline .clang-tidy]: Similar to the other PR, to document why we're doing this, can you add a note above like:

```
  # - readability-redundant-member-init hinders us from setting initializers to
  #   silence -Wmissing-designated-field-initializers.
```

Comment (danakj) [reply .clang-tidy]: Yep, done and I rebased on top of #4540, so it should land first.

**Comment by danakj** [comment]
Rebased, PTAL

**Comment by danakj** [comment]
bump, needs re-merge approval after rebase

**Review by jonmeow — APPROVED** [review]

**Pull request merged**